src/compiler/emitter.ts

```diff
@@ -721,6 +722,62 @@ export function getOutputFileNames(commandLine: ParsedCommandLine, inputFileName
     return getOutputs();
 }
 
+/** @internal */
+export function getSourceMapDirectory(host: EmitHost, mapOptions: SourceMapOptions, filePath: string, sourceFile: SourceFile | undefined) {

```

Comment:
I am currently trying to get rid of the emit host as much as possible since that has a lot of functionality that is not available in the DTE. Could we maybe find a way to subset the methods of `EmitHost` that actually used here? In #124 I manage to get rid of most of the unimplemented parts of emitHost and just keep the core functionality used in source maps:

```ts
  const emitHost = {
        getCurrentDirectory: () => transpileOptions.currentDirectory ?? ".",
        getCanonicalFileName: createGetCanonicalFileName(!!compilerOptions.useCaseSensitiveFileNames),
        useCaseSensitiveFileNames: () => !!compilerOptions.useCaseSensitiveFileNames,
        getCompilerOptions: () => compilerOptions.compilerOptions,
        getCommonSourceDirectory: () => ensureTrailingDirectorySeparator(transpileOptions.commonSourceDirectory ?? "."),
    };
```

Maybe we can do something similar? I seem to remember I used `Pick` in some place to pick out just the actually used methods. Not sure if `Pick` is the best option, but maybe another type with just our methods of interest (or if we find an existing one taht does not grow the API surface too much that would also be great)

Reply:
How about this? I made `EmitHost` -> `CoreEmitHost`